packages/ui/src/memberships/components/MemberListItem/Fileds.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import styled from 'styled-components'
+
+import { Fonts, Sizes } from '../../../common/constants'
+import { Membership, OtherMember } from '../../types'
+
+export const colLayoutByType = (kind: (Membership | OtherMember)['kind']) => {
+  const id = 70
+  const name = 194
+  const concil = 80
+  const roles = 200
+  const count = kind === 'Member' ? 20 : 76
+  const total = 100
+
+  return kind === 'Member'
+    ? `${id}px ${name}px ${concil}px ${roles}px ${count}px ${count}px ${total}px ${total}px`
+    : `${name}px ${roles}px ${count}px ${count}px 96px 76px 54px`
+}
+
+export const Info = styled.span`
+  font-family: ${Fonts.Grotesk};
+  font-weight: 700;
+`
+export const CountInfo = ({ count, times }: { count: number; times?: boolean }) => (
+  <Info>
+    {count}
+    {times && ' times'}
+  </Info>
+)
+
+export const MemberItemWrap = styled.div`
+  display: grid;
+  grid-template-columns: ${({ member }: { member: Membership | OtherMember }) => colLayoutByType(member.kind)};
+  grid-template-rows: 1fr;
+  justify-content: space-between;
+  justify-items: start;
+  align-items: center;
+  width: 100%;
+  height: ${Sizes.accountHeight};
+  padding: 16px 0 16px 14px;
+`
```

Comment:
Instead of adding a type for this, we can have either:

*   two components for this or
*   base component and two extensions for each list type which will define the gaps

This will duplicate some code but would remove a need to have two types, that are not necessary.

I'd go with `MemberItemWrap`:

```
const MemberItemWrap = styled.div`
  display: grid;
  grid-template-rows: 1fr;
  justify-content: space-between;
  justify-items: start;
  align-items: center;
  width: 100%;
  height: ${Sizes.accountHeight};
  padding: 16px 0 16px 14px;
`
const MyMemberItemWrap = styled(MemberItemWrap)`
   grid-template-columns: colLayoutByType('MyMember')
`
const MemberItemWrap = styled(MemberItemWrap)`
   grid-template-columns: colLayoutByType('Member')
`
```

Reply:
I ended up adding a kind prop like you suggested for the colLayoutByType function